### PR TITLE
Add an image collection object to ccdproc

### PR DESCRIPTION
**Pull request by mwcraig**
This doesn't have docs included yet, so not ready to merge, but wanted to get the discussion going.

The docs for this code are currently at http://msum-astro.readthedocs.org/en/latest/image_management.html#image-management if you want to get an overview.

### CODE REVIEW

**Comment by mwcraig** [comment]
This still needs docs, but the tests pass now. If it looks like a direction we want to go I'll write some docs.

**Comment by crawfordsm** [comment]
Question:  Should the default for keywords be '*' ?

File: ccdproc/tests/test_image_collection.py [hunk 1]
```diff
@@ -0,0 +1,539 @@
...
+    # 0.2.4
+    print(np.array(summary['file'] == 'no_filter_no_object_bias.fit'))
+    no_filter_no_object_row = np.array(summary['file'] ==
+                                       'no_filter_no_object_bias.fit')
+    # there should be no filter keyword in the bias file
+    assert (summary['filter'][no_filter_no_object_row].mask)
+
+
+# this should work, but doesn't:
+# @pytest.mark.usefixtures("triage_setup")
```
Comment (crawfordsm) [inline ccdproc/tests/test_image_collection.py]: What's the problem here?   Does it still not work/what is it suppose to do?

File: docs/ccdproc/image_management.rst [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+Selecting files
+---------------
+
+Selecting the files that match a set of criteria, for example all images in
+the I band with exposure time less than 60 seconds you could do::
+
+    >>> matches = (ic1.summary['filter'] == 'I' & ic1.summary['exposure'] < 60)
+    >>> my_files = summary['file'][matches]
+
```
Comment (crawfordsm) [inline docs/ccdproc/image_management.rst]: Add an ic1 before summary

File: docs/ccdproc/image_management.rst [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+argument to the file name between the original base name and extension. The
+example below has (almsot) the same effect of the example above, subtracting
+the mean from each image and saving to a new file::
+
+    >>> for hdu in ic1.hdus(save_with_name='_new',
+    ...                     imagetyp='LiGhT', filter='I', exposure=30):
+    ...    hdu.header['meansub'] = True
+    ...    hdu.data = hdu.data - hdu.data.mean()
+
+   It saves, in the ``location`` of the image collection, a new FITS file with
```
Comment (crawfordsm) [inline docs/ccdproc/image_management.rst]: This ends up a little mangled on build.  Need to adjust the indentiion starting with 'It'.

File: ccdproc/image_collection.py [hunk 4]
```diff
@@ -0,0 +1,643 @@
...
+        for extension in full_extensions:
+            files.extend(fnmatch.filter(all_files, '*' + extension))
+
+        return files
+
+    def _generator(self, return_type,
+                   save_with_name="", save_location='',
+                   clobber=False,
+                   overwrite=False,
+                   do_not_scale_image_data=True,
```
Comment (crawfordsm) [inline ccdproc/image_collection.py]: clobber and overwrite?  Perhaps just use clobber which pyfits uses

File: docs/ccdproc/image_management.rst [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+``save_location``; note that you do not need to actually do anything to the
+hdu (or header or data) to cause the copy to be made. The example below copies
+all of the I-band light files with 30 second exposure from the original
+location to ``"other_dir"``::
+
+    >>> for hdu in ic1.hdus(save_location='other_dir',
+    ...                     imagetyp='LiGhT', filter='I', exposure=30):
+    ...     pass
+
+   This option can be combined with the previous one to also give the files a
```
Comment (crawfordsm) [inline docs/ccdproc/image_management.rst]: Issue with the indentation here

**Comment by crawfordsm** [comment]
Question:  After every iteration, does it have to read/write out the file?  It would probably be much faster if it only had one read/write (although then memory may become an issue)

**Comment by mwcraig** [comment]
Add `pytest-caplog` as a `test_requires` dependency or rewrite tests to not examine log

**Comment by crawfordsm** [comment]
here are the two errors that I am currently getting:

================================================== ERRORS ==================================================
_____________________ ERROR at setup of TestImageFileCollection.test_dir_with_no_keys ______________________
file /private/var/folders/sn/9zbvs5x91t52g_6lq1dn64980000gn/T/ccdproc-test-ne0aXy/lib.macosx-10.5-x86_64-2.7/ccdproc/tests/test_image_collection.py, line 261
      def test_dir_with_no_keys(self, tmpdir, caplog):
        fixture 'caplog' not found
        available fixtures: ccd_data, pytestconfig, recwarn, monkeypatch, triage_setup, capfd, capsys, tmpdir
        use 'py.test --fixtures [testpath]' for help on them.

/private/var/folders/sn/9zbvs5x91t52g_6lq1dn64980000gn/T/ccdproc-test-ne0aXy/lib.macosx-10.5-x86_64-2.7/ccdproc/tests/test_image_collection.py:261
_____ ERROR at setup of TestImageFileCollection.test_initializing_from_table_file_that_does_not_exist ______
file /private/var/folders/sn/9zbvs5x91t52g_6lq1dn64980000gn/T/ccdproc-test-ne0aXy/lib.macosx-10.5-x86_64-2.7/ccdproc/tests/test_image_collection.py, line 412
      def test_initializing_from_table_file_that_does_not_exist(self,
        fixture 'caplog' not found
        available fixtures: ccd_data, pytestconfig, recwarn, monkeypatch, triage_setup, capfd, capsys, tmpdir
        use 'py.test --fixtures [testpath]' for help on them.

/private/var/folders/sn/9zbvs5x91t52g_6lq1dn64980000gn/T/ccdproc-test-ne0aXy/lib.macosx-10.5-x86_64-2.7/ccdproc/tests/test_image_collection.py:412
================================================= FAILURES =================================================
______________________________________ [doctest] image_management.rst ______________________________________
028 You can use the wildcard ``*`` in place of a list to indicate you want the
029 collection to use all keywords in the headers::
030 
031     >>> ic_all = ImageFileCollection('.', keywords='*')
032 
033 Most of the useful interaction with the image collection is via its
034 ``.summary`` property, an `~astropy.table.Table`_ of the value of each keyword for each
035 file in the collection::
036 
037     >>> ic1.summary.colnames
Expected:
    ['imagetyp', 'object', 'filter']
Got:
    ['file', 'filter', 'object', 'imagetyp', 'exposure']

/Users/crawford/programs/tests/test_image/ccdproc/docs/ccdproc/image_management.rst:37: DocTestFailure
============================== 1 failed, 256 passed, 2 error in 13.93 seconds ==============================

File: ccdproc/image_collection.py (hunk 4 above)
Comment (mwcraig) [inline ccdproc/image_collection.py]: Actually, `clobber` is only there for backwards compatibility. I'd prefer to move to overwrite, which seems clearer to the uninitiated.

Comment (bsipocz) [inline ccdproc/image_collection.py]: :+1 to ``overwrite``, ``clobber`` was quite confusing when I first came across with it (in pyfits). Anyone uses pyfits would know what ``overwrite`` means, but it's not true the other way around.

**Pull request merged**